gst/nnstreamer/tensor_common.h

```diff
@@ -126,6 +126,41 @@ typedef enum _nns_media_type
   _NNS_MEDIA_END, /**< End Marker */
 } media_type;
 
+/**
+ * @brief Internal data structure for tensor element value.
+ */
+typedef struct
+{
+  tensor_type type;
+
+  /* tensor element data union for each tensor type */
+  union
+  {
+    TENSOR_TYPE (_NNS_INT32) v_int32;
+    TENSOR_TYPE (_NNS_UINT32) v_uint32;
+    TENSOR_TYPE (_NNS_INT16) v_int16;
+    TENSOR_TYPE (_NNS_UINT16) v_uint16;
+    TENSOR_TYPE (_NNS_INT8) v_int8;
+    TENSOR_TYPE (_NNS_UINT8) v_uint8;
+    TENSOR_TYPE (_NNS_FLOAT64) v_float64;
+    TENSOR_TYPE (_NNS_FLOAT32) v_float32;
+    TENSOR_TYPE (_NNS_INT64) v_int64;
+    TENSOR_TYPE (_NNS_UINT64) v_uint64;
+  } data;
+} GstTensorValue;
```

Comment:
We have
```
/**
 * @brief To make the code simple with all the types. "C++ Template"-like.
 */
typedef union {
  int32_t _int32_t;
  uint32_t _uint32_t;
  int16_t _int16_t;
  uint16_t _uint16_t;
  int8_t _int8_t;
  uint8_t _uint8_t;
  double _double;
  float _float;
  int64_t _int64_t;
  uint64_t _uint64_t;
} tensor_element;
```
in tensor_typedef.h

Reply:
I will chage code to use 'tensor_element', and then move this definition -tensor-value- to tensor-transform